Update iica-catie.csl

POBrien333 opened the pull request:
To continue: https://github.com/citation-style-language/styles/pull/3059

Hi @danirv91
I've made the changes I have found in the guidelines. Could you help and double check I've made all the correct changes?

danirv91 commented:
Hello. Thanks. I updated the style because there are a lot of changes between the current version and the one that I did. and also, this version that I updated is in spanish.  I saw that you deleted for example the text "disponible en", "consultado" and this isn't ok. How can I try the changes? I want to make sure that the changes are still working.

adam3smith commented:
Did you actually try the style out? A lot of the Spanish terms are just generated differently -- handcoding terms into a style the way you did in your submission in any language is generally better avoided. The style above produces URLs as 
Consultado 5 abr. 2016. Disponible en https://github.com/moltude/aspace-patronize.

adam3smith commented:
You can download the style from https://raw.githubusercontent.com/POBrien333/styles/fa760276f34529f009644f9b57899fc2de4ef5ad/iica-catie.csl

danirv91 commented:
Ok, perfect I'm going to check. Thanks

2017-10-23 9:49 GMT-06:00 Sebastian Karcher <notifications@github.com>:

> You can download the style from https://raw.githubusercontent.
> com/POBrien333/styles/fa760276f34529f009644f9b57899f
> c2de4ef5ad/iica-catie.csl
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/citation-style-language/styles/pull/3066#issuecomment-338704360>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AfESRZfwhZ9SNnICb8tsqS1zv18KdHAuks5svLWIgaJpZM4QChQW>
> .
>



-- 
*Daniela Rodríguez Valerio*
Asistente de Biblioteca
Biblioteca Sede Central
Instituto Interamericano de Cooperación para la Agricultura (IICA)
daniela.rodriguez@iica.int
Tel. (+506) 2216-0436
www.iica.int

danirv91 commented:
Hello, I've already checked the style and I found some changes that don't work. I attached a document with the inconsistencies. The term (en línea), is used after the title when we have an associated URL. Except with web sites, with web sites we put (en línea, sitio web)

Also, I created specific styles for book, book chapter without editor, book chapter with editor, article journal, newspaper article, reports, conference proceedings, thesis, web site, report and generic. The rest of them can be entered as generic. 

Thanks
[corrections.docx](https://github.com/citation-style-language/styles/files/1408865/corrections.docx)

danirv91 commented:
Hello, was it possible to fix? Thanks

POBrien333 commented:
@danirv91 
Can you point me to the location in the guidelines where this is explained. I'm not 100% sure that it's the correct way the way you've put it in your .doc, which btw is very handy and well done!

danirv91 commented:
Sure of course. are you referring to the edition? If you do, here you have the guidelines. it is in page number 8: http://www.iica.int/sites/default/files/publications/files/2016/b4013e.pdf

Thanks

POBrien333 commented:
ok.
Problem 1 with books: edition before the editors. I do not see anywhere in the guidelines that it has to be this way round. Unless you can point me to a specific example, I'll not change.
Problem 2 with chapters: It prints the edition with the container-title and not the chapter-title. Printing the edition with the chapter title does not make sense as individual chapters won't have editions, but the whole book that contains this chapter could have. Example here how it prints this:
Nardone, DA. 1990. Collecting and Analyzing Data: Doing and Thinking (en línea). In Walker, HK; Hall, WD; Hurst, JW (eds.). Clinical Methods: The History, Physical, and Laboratory Examinations. 3rd ed. Boston, Butterworths. Consultado 27 dic. 2016. Disponible en http://www.ncbi.nlm.nih.gov/books/NBK353/.

danirv91 commented:
Problem 1: Take a look at this link in page number 13, point b in 2.1.1. Here you can look at the structure of a printed book http://www.iica.int/sites/default/files/publications/files/2016/b4013e.pdf
"Edición" is edition number and "Mención de autor secundario is where you can put the editors in case.

Problem 2: Look at the page number 25, point 2.6.2.2. Here you have the reference of a book chapter when the author of the chapter isn't the editor of the book. It's true that it doesn't show how to put the edition, because is a big exception. But in page number 8, it says that the edition will always be after the title of the specific information. But anyway as I said it is a big exception so let's go ahead, we can survive without the edition in this case. Eventually, people can add it manually.

Thanks

POBrien333 commented:
That's issue 1 fixed. 
@rmzelle 
I think this can be merged now.

danirv91 commented:
Hi thanks. Issue 1 is perfect now. But I tried the style and it stills having a few troubles. I attached them. And I have a doubt: Why I'm not as a collaborator? I spent a lot of time editing the style with the visual editor from Mendeley and trying it. That's how I managed to edit the file of Marcelo Wall in order to upload a new one.

Thanks
[correcciones3.docx](https://github.com/citation-style-language/styles/files/1441969/correcciones3.docx)

POBrien333 commented:
@danirv91 
check out the new style. Let me know what's missing.

danirv91 commented:
Hello everything is almost perfect now. Thank you very much. I've found only one detail with newspapers: This is the order of the date: day month, not month day. For example, right now, it is like this: nov. 26, but the correct way is 26 nov. You can check it in page number 18 (point 2.2.3.4) in the guidelines. 

and I found other detail with reports. The pages are after the publisher, then it goes the title of the serie and then the number of the serie. Look at this in page 21-22 of the guidelines:

Example:

Pérez, L. (2011). Plagas de la zanahoria (en línea). San José, Costa Rica, IICA.  **230 p. (Reporte técnico). Informe técnico n.° 310**. Consultado 3 oct. 2017. Disponible en https://eravila.wordpress.com/?s=Mendeley.

and the "generic" is missing the country, the publisher and the correct way of the pages. Here is an example:

Pérez, P. (2015). Animales productores de leche (en línea). **City, country, publisher. 89 p**. Consultado 5 feb. 2016. Disponible en http://www.oie.int/wahis_2/wah/health_v7_en.php 

Thanks :)

POBrien333 commented:
article-newspaper: date issue: fixed
report: added genre and collection-title/number (let me know if this works now)
generic: you need to enter the country in zotero yourself. in csl there is only one variable (publisher-place), so enter the location as you want, e.g. "San Jose, Costa Rica" and not just "San Jose"

danirv91 commented:
I've checked the issue with newspapers and now is working perfect.

Report: Something is happening now, when I tried to enter a report, Mendeley and Word had a crash. This is only happenning with reports.

Generic: I'm checking with Mendeley and I already have the information about the country and publisher, but it isn't working. and the page remain in an incorrect way, this is the format: 80 p.

adam3smith commented:
The crash definitely has nothing to do with the citation style. If it persists, report to Mendeley. 

For Generic -- the contents of the "Country" field in Mendeley are irrelevant. CSL never sees them. You need to include the whole info ("San Jose, Costa Rica", in this case) in the City field.

The style correctly prints number of pages as 80 p. Mendeley incorrectly maps this to pages (which is for page ranges and thus correctly displayed as p. 80).

rmzelle commented:
> The style correctly prints number of pages as 80 p. Mendeley incorrectly maps this to pages

@adam3smith, is that for the Generic item type? I don't see "number-of-pages" mentioned at all at https://service.elsevier.com/app/answers/detail/a_id/22175/supporthub/mendeley/.

adam3smith commented:
right, that was an abbreviated account -- Mendeley just does not have number-of-pages at all and always uses pages and always maps that to pages in CSL which makes no sense for this example here.

danirv91 commented:
Hi, I attached an example about what is exactly happening with "generic". 
[correcciones 4.docx](https://github.com/citation-style-language/styles/files/1450713/correcciones.4.docx)

Regarding the "reports", I understood that it isn't related with the citation style. But I haven't been able to generate a citation, it stills crashing, and it's the same with "working papers". It's strange because both of them have the same structure, and precisely, are the only ones that are giving trouble.

adam3smith commented:
On generic, re-read what I wrote. I'm not sure how I can express this more clearly:

> the contents of the "Country" field in Mendeley are irrelevant. CSL never sees them. You need to include the whole info ("San Jose, Costa Rica", in this case) in the City field.

danirv91 commented:
yes I understood, but why is it working perfectly with books for example? Look at the attachment. 
![book](https://user-images.githubusercontent.com/32576069/32506148-76168fd8-c3a9-11e7-8db5-65c576331c96.JPG)

and then look at the reference, it is perfect:

FAO; IICA. 2017. Soils’ potential to contribute to offset international aviation emissions (en línea). 3 ed. Sáenz, A (ed.). San José, Costa Rica, s.e. 30 p. Consultado 18 sep. 2017. Disponible en http://opackoha.iica.int/.

adam3smith commented:
I don't know. That's really up to Mendeley though. They do seem to tag the Country field onto the the City field in some cases, but we have no influence on that in the citation styles, so it's not something we can solve here.

danirv91 commented:
ok But I keep don't understanding why is it possible with books, and isn't with generic. But well, let's do as you said, but if you noticed, the city field is not showing in generic, neither the publisher...

adam3smith commented:
ah I didn't catch that City was missing, too, that's different. @POBrien333 -- "generic" in Mendeley is `article` in CSL, so you should be able to add that to the condition printing publisher&place.

POBrien333 commented:
thanks for the troubleshooting, guys.
Daniela, made an edit. Can you test that? I think that should work.

danirv91 commented:
Generic: now is almost perfect. Only the pages are putting twice. Look:


Pérez, P. 2015. Animales productores de leche (en línea). San José, costa rica, IICA. **p. 89:89**. Consultado 5 feb. 2016. Disponible en https://eravila.wordpress.com/?s=Mendeley.

and it is in this way:


Pérez, P. 2015. Animales productores de leche (en línea). San José, costa rica, IICA. **89 p**. Consultado 5 feb. 2016. Disponible en https://eravila.wordpress.com/?s=Mendeley.

Regarding the "reports" and "working papers", I even desinstalled the Mendeley and the trouble persist, **only** with them. I don't know what to do, I even wrote to Mendeley.

adam3smith commented:
@POBrien333  you have the page number in locators and then again in the bibliography -- I'd just see if you can include "article" somewhere else in the bibliography, e.g. with books or reports?

Sorry, I'd do this myself but there's been so much back & forth on this that I don't want to implement something without thoroughly testing.

There's absolutely nothing in this style that should lead Mendeley to crash -- not even something particularly unusual. You'll have to get them to fix that, I'm afraid, though if they take too long we can try to get the attention of a senior dev there.

POBrien333 commented:
thanks. let's see if this one works. (In Zotero article maps to Document, no? why can't I input a publisher-location or number-of-pages?)

adam3smith commented:
No, Zotero doesn't use article. Document is unmapped (which is why we tell people not to use it). The only way to use it is to not test for type or in an else condition. If you want to test output for article in Zotero, pick whatever item type you want and add `type: article` in Extra.

POBrien333 commented:
I can't reproduce the page 89 error, you are getting. I just tried it with this citation and it works.

@book{perez_animales_2009,
	address = {San José, Costa Rica},
	title = {Animales productores de leche},
	url = {https://eravila.wordpress.com/?s=Mendeley.},
	urldate = {2017-02-02},
	publisher = {IICA},
	author = {Pérez, P},
	year = {2009},
	note = {type: article}
}

Re the report issue: Adam, why is this style actually testing for an event (line 303)? To test if it was a paper-conference or speech which happened at an event? I could pull the report out of this as it's own else-if and see what happens then?

adam3smith commented:
> why is this style actually testing for an event (line 303)? To test if it was a paper-conference or speech which happened at an event

Yeah, that's how I would interpret it. Might be cleaner to pull report out. 

Haven't tested the generic code, but it looks like you don't have a page in there? try with


```
type: article 
page: 89
```

in extra --mind the new line

POBrien333 commented:
ohh. the bitex doesn't show the "number-of-pages" i had in there.
It comes out like this for me: 
Pérez, P. 2009. Animales productores de leche (en línea) (en línea). San José, costa rica, IICA. 89 p. Consultado 2 feb. 2017. Disponible en https://eravila.wordpress.com/?s=Mendeley.

![image](https://user-images.githubusercontent.com/954979/32627487-debf5ec6-c592-11e7-81e3-eafc96fe16f8.png)

adam3smith commented:
since Mendeley doesn't have number-of-pages, I'm guessing they're using page instead and that gets duplicated.

danirv91 commented:
Thanks, now the page number is working perfectly. I tried the conference proceedings and they have a weird structure but you can put them in the category of generic, to make it simple.

The problem with reports and working papers persists I don't know why. I tried with APA style and Mendeley don't crash, also I tried with the style I've created and it works too.

POBrien333 commented:
Glad to hear one problem is solved.
Re Conference-papers: can you point me to the correct location of the guidelines for those? I'll delete them and make them new. Maybe that removes the error.

danirv91 commented:
Sure, page number 26. Point 2.7.1. Thanks

danirv91 commented:
Thanks. Now is like this:

Gómez, S. 2013. La carne vacuna (en línea). . In (2013). Méndez, S (ed.). Actas de la conferencia sobre ganado.  Anais... . San José, Costa Rica, IICA. pp. 83-89. Disponible en http://csl.mendeley.com/logout.php.

Could it be like this?

Gómez, S. 2013. La carne vacuna (en línea). In Actas de la conferencia sobre ganado. Méndez, S (ed.).   San José, Costa Rica, IICA. p. 83-89. Disponible en http://csl.mendeley.com/logout.php.

Thanks

danirv91 commented:
Hello, now is like this:

Gómez, S. 2013. La carne vacuna (en línea). . In Actas de la conferencia sobre ganado.  Anais... . Méndez, S (ed.). San José, Costa Rica, IICA. p. 83-89. Disponible en http://csl.mendeley.com/logout.php.

Only remove "anais":

Gómez, S. 2013. La carne vacuna (en línea). . In Actas de la conferencia sobre ganado.  **Anais... .** Méndez, S (ed.). San José, Costa Rica, IICA. p. 83-89. Disponible en http://csl.mendeley.com/logout.php.

Thanks

POBrien333 commented:
What field is this Anaïs bit in your Zotero entry?

danirv91 commented:
Hello, anais ir a word in portuguese, used to refer to an annual publication. But is only used if the meeting procedding doesn't have a title and you want to put "something generic". That's a very very specific case, and we commonly have the titles.

adam3smith commented:
The question is where you have included that word in Mendeley (in which field) 

@POBrien333 also note the duplicate punctuation after the title

danirv91 commented:
ahh, no I didn't include that word. It appears automatically. So I suppose is part of the style. and yes you are right with the double period.

adam3smith commented:
@POBrien333 -- line 64, hardcoded as text value

OK, so this currently appears for all conference papers -- is that correct? Or should it only appear where there is no proceedings title?

danirv91 commented:
currently is not correct, is only when there's no proceedings title. and this term anais is in portuguese. Also, putting the term "anais" or the equal in spanish is not totally correct. The guidelines explain that you have to put something instead of the title according to the content, but is not necessarily anais. Look the last paragraph on page 23 http://www.iica.int/sites/default/files/publications/files/2016/b4013e.pdf

adam3smith commented:
OK @POBrien333 I'd delete the hard-coded term then and replace it by genre *where there is no container-title*

danirv91 commented:
thanks. Only delete the double puntuaction after the title and after the container title:

Gómez, S. 2013. La carne vacuna (en línea). **.** In Actas de la conferencia sobre ganado. **.** Méndez, S (ed.). San José, Costa Rica, IICA. p. 83-89. Disponible en http://csl.mendeley.com/logout.php.

I tried both cases (with title and without title, putting the title in genre) and it works good.

danirv91 commented:
The point after the container title is still there.

Gómez, S. 2013. La carne vacuna (en línea). In Actas de la conferencia sobre ganado. . Méndez, S (ed.). San José, Costa Rica, IICA. p. 83-89. Disponible en http://csl.mendeley.com/logout.php.

POBrien333 commented:
I honeslty don't know where that extra point is coming from. Can you double check this with another conference paper?

It comes out like this for me: 
Alqueres, JL. 1991. The Brazilian Power System and the challenge of the Amazon transmission (en línea). Praca, JC (ed.). Miracletown, IEEE. p. 315-320 DOI: https://doi.org/10.1109/TDC.1991.169526.

danirv91 commented:
Your reference is perfect. I tried with another one in Mendeley and the point is still there.

POBrien333 commented:
I'm at a loss here. Can you give me a link to one of your citations, or if that is not possible post the bibtex of it here, so I can import it into Zotero and Mendeley and troubleshoot further?

danirv91 commented:
Sure, Look, I created another example with this one: https://goo.gl/8Sp5wv

Also, I used the Mendeley web importer to reference the information in page 21

I've attached a screenshot with the information in the web importer and also the bibtex

[proceeding.zip](https://github.com/citation-style-language/styles/files/1476122/proceeding.zip)


Thanks

POBrien333 commented:
Thanks for those citations. very useful for future style makings.
Also found the error. I wasn't thinking straight.... Perks of starting a new job.

danirv91 commented:
You are welcome. and everything is perfect now. The only issue that keeps me worrying, is the issue with reports and working papers. I didn't manage to review that, they don't work.

adam3smith commented:
OK, I'm going to accept this then. FWIW, I cannot replicate the issue with reports in Mendeley.

POBrien333 commented:
Thanks.
#3059. An be closed then.

danirv91 commented:
Sorry which version of Mendeley are you using? me and other 3 people tried in different computers and they can't produce reference for working papers and reports as well. Maybe it could be the version..

adam3smith commented:
I tested on an old 1.15.3 for Windows.

danirv91 commented:
Yes, I've installed that old version and everything work. But I don't know how can we solve that. People tend to download the last version or they install the updates.

Also, I saw a problem:


Pérez, L. (2011). Plagas de la zanahoria (en línea). 3. San José, Costa Rica, IICA. Reporte técnico. Consultado 3 oct. 2017. Disponible en https://eravila.wordpress.com/?s=Mendeley.

The number 3 is the series number, and it is located after the series name, like this:

Pérez, L. (2011). Plagas de la zanahoria (en línea). San José, Costa Rica, IICA. Reporte técnico. 3. Consultado 3 oct. 2017. Disponible en https://eravila.wordpress.com/?s=Mendeley.

danirv91 commented:
hello?

danirv91 commented:
Hello, can someone tell me what's the next step with this issue?
Thanks
